src/DepthImageToLaserScanROS.cpp

```diff
@@ -27,65 +27,88 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-/* 
+/*
  * Author: Chad Rockey
  */
 
+#include <rclcpp/rclcpp.hpp>
+#include <sensor_msgs/msg/camera_info.hpp>
+#include <sensor_msgs/msg/image.hpp>
+#include <sensor_msgs/msg/laser_scan.hpp>
+
+#include <functional>
+
 #include <depthimage_to_laserscan/DepthImageToLaserScanROS.h>
 
+#define ROS_ERROR printf
+
 using namespace depthimage_to_laserscan;
-  
-DepthImageToLaserScanROS::DepthImageToLaserScanROS(ros::NodeHandle& n, ros::NodeHandle& pnh):pnh_(pnh), it_(n), srv_(pnh) {
-  boost::mutex::scoped_lock lock(connect_mutex_);
-  
-  // Dynamic Reconfigure
-  dynamic_reconfigure::Server<depthimage_to_laserscan::DepthConfig>::CallbackType f;
-  f = boost::bind(&DepthImageToLaserScanROS::reconfigureCb, this, _1, _2);
-  srv_.setCallback(f);
-  
-  // Lazy subscription to depth image topic
-  pub_ = n.advertise<sensor_msgs::LaserScan>("scan", 10, boost::bind(&DepthImageToLaserScanROS::connectCb, this, _1), boost::bind(&DepthImageToLaserScanROS::disconnectCb, this, _1));
+
+DepthImageToLaserScanROS::DepthImageToLaserScanROS(rclcpp::node::Node::SharedPtr & node):node_(node) {
+  rmw_qos_profile_t custom_qos_profile = rmw_qos_profile_default;
```

Comment:
I know it's redundant in all my reviews but please build your custom QOS profiles on top of the `rmw_qos_profile_sensor_data` profile that is designed exactly for this purpose